Include/internal/pycore_abstract.h

```diff
@@ -8,6 +8,18 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
+/// Return 1 if the specified index is in the range [0, limit)
+static inline int _Py_is_valid_index(Py_ssize_t i, Py_ssize_t limit)
+{
```

Comment:
You should mention that this function relies on twos-complement representation, and is not affected by -fwrapv, -fno-wrapv and -ftrapv compiler options of GCC and clang.